bundles/org.openhab.automation.jythonscripting/src/main/java/org/openhab/automation/jythonscripting/JythonScriptEngineFactory.java

```diff
@@ -0,0 +1,132 @@
+import org.osgi.service.component.annotations.Deactivate;
+
+/**
+ * This is an implementation of {@link ScriptEngineFactory} for Jython.
+ *
+ * @author Scott Rushworth - Initial contribution
+ * @author Wouter Born - Initial contribution
+ */
+@Component(service = ScriptEngineFactory.class)
+@NonNullByDefault
+public class JythonScriptEngineFactory extends AbstractScriptEngineFactory {
+
+    private static final String PYTHON_CACHEDIR = "python.cachedir";
+    private static final String PYTHON_HOME = "python.home";
+    private static final String PYTHON_PATH = "python.path";
+
+    private static final String DEFAULT_PYTHON_PATH = Paths
+            .get(OpenHAB.getConfigFolder(), "automation", "lib", "python").toString();
+
+    private static final String SCRIPT_TYPE = "py";
+    private static final javax.script.ScriptEngineManager ENGINE_MANAGER = new javax.script.ScriptEngineManager();
+
+    @Activate
+    public JythonScriptEngineFactory() {
+        logger.debug("Loading JythonScriptEngineFactory");
+
+        String pythonHome = JythonScriptEngineFactory.class.getProtectionDomain().getCodeSource().getLocation()
+                .toString().replace("file:", "");
+        System.setProperty(PYTHON_HOME, pythonHome);
+
+        String existingPythonPath = System.getProperty(PYTHON_PATH);
+        if (existingPythonPath == null || existingPythonPath.isEmpty()) {
+            System.setProperty(PYTHON_PATH, DEFAULT_PYTHON_PATH);
+        } else if (!existingPythonPath.contains(DEFAULT_PYTHON_PATH)) {
+            Set<String> newPythonPathList = new TreeSet<>(
+                    new ArrayList<>(Arrays.asList(existingPythonPath.split(File.pathSeparator))));
+            newPythonPathList.add(DEFAULT_PYTHON_PATH);
+            String newPythonPath = String.join(File.pathSeparator, newPythonPathList);
+            System.setProperty(PYTHON_PATH, newPythonPath);
+        }
+
+        System.setProperty(PYTHON_CACHEDIR,
+                Paths.get(OpenHAB.getUserDataFolder(), "cache", "org.openhab.automation.jythonscripting", "cachedir")
+                        .toString());
+
+        logPythonPaths();
+    }
+
+    private void logPythonPaths() {
+        logger.trace("{}: {}, {}: {}, {}: {}", //
+                PYTHON_HOME, System.getProperty(PYTHON_HOME), //
+                PYTHON_PATH, System.getProperty(PYTHON_PATH), //
+                PYTHON_CACHEDIR, System.getProperty(PYTHON_CACHEDIR));
+    }
+
+    @Override
+    public List<String> getScriptTypes() {
+        List<String> scriptTypes = new ArrayList<>();
+
+        for (javax.script.ScriptEngineFactory factory : ENGINE_MANAGER.getEngineFactories()) {
+            List<String> extensions = factory.getExtensions();
+
+            if (extensions.contains(SCRIPT_TYPE)) {
+                scriptTypes.addAll(extensions);
+                scriptTypes.addAll(factory.getMimeTypes());
+            }
+        }
+        return List.copyOf(scriptTypes);
```

Comment:
Not sure why you would need to copy it.
```suggestion
        return scriptTypes;
```

Reply:
Applied because it will generate a new result any time so immutability isn't required.